src/components/organisms/Room/Room.tsx

```diff
@@ -210,24 +116,18 @@ const Room: React.FC<RoomProps> = ({
     [hasChairs, participants.length, participantContainerClassName]
   );
 
-  if (!token) {
-    return <></>;
+  if (loading) {
+    return <Loading />;
   }
 
   return (
     <Fragment>
       {meComponent}
       {othersComponents}
       {emptyComponents}
-      <VideoErrorModal
-        show={!!videoError}
-        onHide={() => setVideoError("")}
-        errorMessage={videoError}
-        onRetry={connectToVideoRoom}
-        onBack={() => (setSeatedAtTable ? leaveSeat() : setVideoError(""))}
-      />
+      {renderErrorModal((dismissVideoError: () => void) =>
+        setSeatedAtTable ? leaveSeat() : dismissVideoError
+      )}
```

Comment:
It looks junky. Not only it is located in `return`, but it also has strange logic. If bool is true, call a function, if not - return a function

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2369/commits/8fa06206d8a463463d85380f61fff2b4dd953adb